docs/en_us/developers/source/accessibility.rst

```diff
@@ -0,0 +1,547 @@
+Alt attributes
+--------------
+
+Regardless of whether or not an image requires a text alternative, *all*
+``<img>`` elements require an ``alt`` attribute, even if the value of that
+attribute is ``""`` (NULL). If your image is purely decorative, or has a text
+alternative immediately adjacent to it, use a NULL alt attribute: ``alt=""``.
+
+If an ``<img>`` element does not have a NULL alt attribute, you should make sure
+that the value you use in its alt attribute provides useful information to users
+who rely on screen readers. Screen readers will expose the path to the image as
+a last resort, if an alt attribute value does not exist.
+
+
+------------------------------------
+Best practices for non-text elements
+------------------------------------
+
+Providing *useful* text alternatives or alt attribute values is more difficult
+than it sounds. Ask yourself questions about the purpose of your image to
+determine what would be most useful to the user.
+
+* Is your image the only content of a link or form control?
+
+    Your ``alt`` attribute should describe the destination of the link, or the
+    action that will be performed. For example, a "Play" icon should have a text
+    alternative such as "Play the 'Introduction to Linux' course video", rather
+    than "Right-pointing triangle".
+    
+* Does your image contain text? The vast majority of images of text should
+  include the verbatim text as the value of the ``alt`` attribute. Here are some
+  examples of exceptions.
+  
+    * If yes, and if the same text appears adjacent to or near the image in the
+      DOM, use a "NULL" value in the alt attribute, otherwise a screen reader is
+      exposed to the same content twice.
+    
+    * If yes, and if the text within the image is there simply for visual effect
+      (such as a skewed screenshot of computer code), use a "NULL" value in the
+      alt attribute.
+	  
+* Does your image contribute meaning to the current page or context?
+  
+    * If yes, and if the image is a simple graphic or photograph, the alt
+      attribute should briefly describe the image in a way that conveys the same
+      meaning that a sighted person would obtain from viewing the image. Context
+      is important. A detailed description of a photograph is rarely useful to
+      the user, unless it is in the context of a photography or art class.
+	  
+    * If yes, and if the image is a graph or complex piece of information,
+      include the information contained in the image elsewhere on the page. The
+      alt attribute value should give a general description of the complex image. You
+      can programmatically link the image with the detailed information using ``aria-
+      describedby``.
+	
+A very pragmatic guide on providing useful text alternatives is included in the
+`HTML5 specification (4.7.1.1) <http://www.w3.org/TR/html5/embedded-
+content-0.html#alt>`_ and includes a variety of example images and appropriate
+text alternatives.
```

Comment:
Can we add "A more comprehensive decision tree is available in the `Web Accessibility Initiatives Images Tutorial <http://www.w3.org/WAI/tutorials/images/>`